### ISSUE TITLE
Alloc output without compiling a Fusion

### ISSUE THREAD

**Issue by samnordmann** [NVIDIA/Fuser#1526]
In this PR about multi-device: https://github.com/NVIDIA/Fuser/pull/1244
we need to allocate some intermediate tensor buffer. The only simple way to do it right now is as follow:
- copy the fusion and change the output to be the tensors we need to allocate
- compile the fusion
- call the method allocOutputSpace

Copying and compiling the Fusion is wasteful here. We should refactor FusionExecutor and expose a utility "allocTensors" that do not need the Fusion compilation.

**Comment by kevinstephano** [comment]
cc @wujingyue .  Jingyue has been thinking about this as well for his work in segmentation with respect to decoupling allocoations and specific kernels but that is not related to multi-device usage.  It would be good if everyone is aligned!

**Comment by wujingyue** [comment]
Yeah, meta-op-only fusions go through the same unnecessary fusion-to-kernel compilation. It's fortunately less wasteful because I changed `expr_sort.cpp` to [not put alias-computing `Expr`s](
https://github.com/NVIDIA/Fuser/blob/101530ec09a7eab9937f277715be0e3ecc1478b6/csrc/device_lower/pass/expr_sort.cpp#L1509-L1510) to the kernel IR.

**Comment by naoyam** [comment]
@wujingyue Is this still relevant?
